misc/palindrome.c

```diff
@@ -1,26 +1,39 @@
+/**
+ * @file
+ * @brief Program to identify if a number is [palindrome
+ * number](https://en.wikipedia.org/wiki/Palindrome) or not.
+ */
```

Comment:
```suggestion
 * @see project_euler/problem_4/sol1.c
 */
```
This will provide an auto-link from this file doc to the other file. You can add a similar line 
`@see misc/palindrome.c` in the main doc of the other file for symmetry

Reply:
@kvedala Please review again